docs/resources/ec_deployment.md

```diff
@@ -349,6 +380,16 @@ The optional `elasticsearch.trust_external` block, allows external trust relatio
 * `trust_all` (Optional) If true, all clusters in this external entity will be trusted and the `trust_allowlist` is ignored.
 * `trust_allowlist` (Optional) The list of clusters to trust. Only used when `trust_all` is `false`.
 
+##### Strategy
+
+The optional `elasticsearch.strategy` allow to choose the configuration strategy used to apply the changes.
```

Comment:
IMO we should have something in here along the lines of "don't change this unless you know that you need to", it's not functionality customers should be concerned with.

Reply:
I have added kind of a warning.